detox/index.d.ts

```diff
@@ -576,6 +583,12 @@ declare global {
              */
             uninstallApp(bundle?: string): Promise<void>;
 
+            /**
+             * Reset the state of the app (e.g. by reinstalling on the device)
+             * or deleting user data.
+             */
+            resetAppState(): Promise<void>;
```

Comment:
@jonathanmos @d4vidi @asafkorem 

I suggest renaming to `clearAppData`. It is more coherent with the UI terminology in iOS and Android:

![image](https://user-images.githubusercontent.com/1962469/161954297-4be6ff85-2014-4162-b928-89bf27b18170.png)

![image](https://user-images.githubusercontent.com/1962469/161954336-dde470d7-9fe2-4697-af95-e49b7b25fa4e.png)

and Google also suggests it to be less popular 

![image](https://user-images.githubusercontent.com/1962469/161954393-aae36df5-a9b3-4aa3-b8b7-ee298287909f.png)

![image](https://user-images.githubusercontent.com/1962469/161954416-2ac7db3b-01bc-4f28-b972-0d23ff7391d4.png)

Reply:
I think that resetting app state is _what_ this method does, and clearing data is _how_ it does it (when optimized on Android, at least). So, I vote for leaving as-is 😅

Reply:
It is not app state. Reset app state is more like something that happens in runtime, via Redux or other state management library.